cmake/compiler_msvc.cmake

```diff
@@ -0,0 +1,13 @@
+# MSVC
+
+if(MSVC)
```

Comment:
This `if(MSVC)` is unnecessary, since the code only gets included if MSVC was already true in the `compilers.cmake` fragment.

Also, you're not using the `${LIST_INCLUDE_DIRS}` variable, are you sure you don't need it for MSVC?

If the generic approach works, simply add `include_directories(${LIST_INCLUDE_DIRS}`, but on gcc/clang for example its better to pass them using `-isystem` so they get treated as system code (warnings deactivated), I don't know if something like this makes sense on MSVC too.